spec/controllers/idv/scan_id_controller_spec.rb

```diff
@@ -0,0 +1,149 @@
+require 'rails_helper'
+
+describe Idv::ScanIdController do
+  before do |test|
+    stub_sign_in unless test.metadata[:skip_sign_in]
+    allow(Figaro.env).to receive(:enable_mobile_capture).and_return('true')
+  end
+
+  describe '#new' do
+    it 'works' do
+      get :new
+    end
+
+    context 'no user signed in', :skip_sign_in do
+      it 'redirects to the root url with no token' do
+        get :new
+
+        expect(response).to redirect_to root_url
+      end
+
+      it 'redirects to the root url with a bad token' do
+        get :new, params: { token: 'foo' }
+
+        expect(response).to redirect_to root_url
+      end
+
+      it 'redirects to the root url with a bad token' do
+        get :new, params: { token: 'foo' }
+
+        expect(response).to redirect_to root_url
+      end
+
+      it 'works with a good token' do
+        token = CaptureDoc::CreateRequest.call(create(:user).id).request_token
+        get :new, params: { token: token }
+
+        expect(response).to render_template(:new)
+      end
+    end
+  end
+
+  describe '#scan_complete' do
+    it 'works' do
+      get :scan_complete
+    end
+
+    it 'works when all the checks pass' do
+      controller.user_session['idv/doc_auth_v2'] = {}
+      session[:scan_id] = { instance_id: 'foo', liveness_pass: true, facematch_pass: true, pii: {} }
+      get :scan_complete
+      expect(response).to redirect_to(idv_doc_auth_v2_step_url(step: :ssn))
+    end
+
+    it 'works when all the checks pass in hybrid flow', :skip_sign_in do
+      token = CaptureDoc::CreateRequest.call(create(:user).id).request_token
+      get :new, params: { token: token }
+
+      session[:scan_id] = { instance_id: 'foo', liveness_pass: true, facematch_pass: true, pii: {} }
+      get :scan_complete
+      expect(response).to render_template(:capture_complete)
+    end
+  end
+
+  describe '#field_image' do
+    it 'works' do
```

Comment:
Are we going to add more thorough tests, for actual behavior?

Reply:
Started working on feature specs but ran into a snag.  @jmhooper is in the process of hooking me up.  For some reason simply adding js wasn't enough.

Reply:
ah forgot it was the wasm

Reply:
in a controller spec, I don't expect wasm would be an issue? These tests should be testing form posts, etc etc right?

Reply:
I though u mentioned feature specs.  When u mentioned testing behavior there was almost 100% coverage.

Reply:
Ok maybe I tagged the wrong line... this is not a meaningful amount of coverage to me:

```ruby
    it 'works' do
      get :new
    end
```

because it tells us that yes, the `#new` method exists and doesn't blow up, but we don't confirm anything about the inputs or outputs or renders or anything

Reply:
ah ok...no problemo!

Reply:
ok let me know how the last push works for you.

Reply:
Yeah it's way clearer, thank you!